### PR TITLE
scheduler: support loose store picking for hot region scheduling

### PR DESCRIPTION
**Pull request by Yisaer**
### What is changed and how it works?

Support picking store with loose condition

### Check List

Tests

- Unit test

### Release note

```release-note
Support `strict-picking-store` in hot region scheduler in order to control whether picking store with strict condition.
```

### CODE REVIEW

File: server/schedulers/hot_region.go [hunk 1]
```diff
@@ -593,12 +593,21 @@ func (bs *balanceSolver) filterSrcStores() map[uint64]*storeLoadDetail {
...
+			return true
+		})
+	}
+	switch bs.firstPriority {
+	case stringToDim(BytePriority):
+		return minLoad.Loads[statistics.ByteDim] > bs.sche.conf.GetSrcToleranceRatio()*expectLoad.Loads[statistics.ByteDim]
+	case stringToDim(KeyPriority):
+		return minLoad.Loads[statistics.KeyDim] > bs.sche.conf.GetSrcToleranceRatio()*expectLoad.Loads[statistics.KeyDim]
+	}
+	return false
```
Comment (lhy1024) [inline server/schedulers/hot_region.go:610]:
```suggestion
	return minLoad.Loads[bs.firstPriority] > bs.sche.conf.GetSrcToleranceRatio()*expectLoad.Loads[bs.firstPriority]
```

File: server/schedulers/hot_region.go [hunk 2]
```diff
@@ -784,12 +793,21 @@ func (bs *balanceSolver) pickDstStores(filters []filter.Filter, candidates []*st
...
-	})
+	if bs.sche.conf.StrictPickingStore {
+		return slice.AllOf(maxLoad.Loads, func(i int) bool {
+			if bs.isSelectedDim(i) {
+				return maxLoad.Loads[i]*dstToleranceRatio < expect.Loads[i]
+			}
+			return true
+		})
+	}
+	switch bs.firstPriority {
```
Comment (lhy1024) [inline server/schedulers/hot_region.go:804]: ditto

**Review by lhy1024 — APPROVED** [review]

File: server/schedulers/hot_region_config.go [hunk 3]
```diff
@@ -82,6 +83,7 @@ type hotRegionSchedulerConfig struct {
 	DstToleranceRatio      float64  `json:"dst-tolerance-ratio"`
 	ReadPriorities         []string `json:"read-priorities"`
 	WritePriorities        []string `json:"write-priorities"`
+	StrictPickingStore     bool     `json:"strict-picking-store"`
```
Comment (nolouch) [inline server/schedulers/hot_region_config.go:86]: I think it should be `json:"strict-picking-store,string"`, Do you test it with `pd-ctl`? BTW, It required 2LGT.

**Pull request merged**